### PR TITLE
Add github/cli/cli

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by scopatz** [comment]
LGTM

File: recipes/gh/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  build:
+    - make     # [unix]
+    - m2-make  # [win]
+    - {{ compiler('go') }}
+
+test:
+  requires:
+    - m2-bash  # [win]
+  commands:
+    - bash -c "{{ name|lower }} --version"
```
Comment (isuruf) [inline recipes/gh/meta.yaml]: Why do you need a `bash` call here?

Comment (mariusvniekerk) [reply recipes/gh/meta.yaml]: Mostly copied from the other golang recipes

File: recipes/gh/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
```
Comment (isuruf) [inline recipes/gh/meta.yaml]:
```suggestion
    - {{ name|lower }} --version
```

File: recipes/gh/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+requirements:
+  build:
+    - make     # [unix]
+    - m2-make  # [win]
+    - {{ compiler('go') }}
+
+test:
+  requires:
+    - m2-bash  # [win]
```
Comment (isuruf) [inline recipes/gh/meta.yaml]: You can remove the two lines above.

**Pull request closed without merging**